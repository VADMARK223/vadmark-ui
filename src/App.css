.btn {
    color: #0074D9;
    background: #440044;
}