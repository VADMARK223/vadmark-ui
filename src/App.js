/**
 * @author Markitanov Vadim
 * @since 21.01.2023
 */
import {useEffect, useState} from "react";

const App = (props) => {
    const [data, setData] = useState(null);
    const [toggle, setToggle] = useState(false);
    const [value, setValue] = useState('');

    useEffect(()=> {
        setTimeout(()=>{
            setData({});
        }, 100);
    }, []);

    const handlerClick = ()=> {
        setToggle(!toggle);
    }

    return (
        <div>
            <div data-testid={'input-value'}>{value}</div>
            {toggle === true && <div data-testid={'toggle-div-id'}>Toggle</div>}
            {data && <div>Data</div>}
            <h1>Hello world</h1>
            <button data-testid={'toggle-button-id'} onClick={handlerClick}>Click me</button>
            <input data-testid={'input-id'} type={'text'} placeholder={'input value..'} onChange={e=>setValue(e.target.value)}/>
        </div>
    )
}

export default App;