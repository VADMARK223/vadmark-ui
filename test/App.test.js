/**
 * @author Markitanov Vadim
 * @since 21.01.2023
 */
import {render, screen, fireEvent} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from "../src/App";
import '@testing-library/jest-dom';

describe('App tests', () => {
    test('Basic test', () => {
        // const component = renderer.create(<App/>);

        render(<App/>);
        const helloWorld = screen.getByText(/hello world/i);
        const button = screen.getByRole('button');
        const input = screen.getByPlaceholderText(/input value/i)

        expect(helloWorld).toBeInTheDocument();
        expect(button).toBeInTheDocument();
        expect(input).toBeInTheDocument();
        expect(input).toMatchSnapshot();
        // screen.debug();
    });

    test('Basic test 2', () => {
        render(<App/>);
        const helloWorld = screen.queryByText(/hello2/i);
        // const helloWorld = screen.getByText(/hello2/i); // Сразу кидает исключение, если не  находит
        expect(helloWorld).toBeNull();
    });

    test('Basic test 3', async () => {
        render(<App/>);
        // screen.debug();
        const helloWorld = await screen.findByText(/data/i);
        expect(helloWorld).toBeInTheDocument();
        // screen.debug();
    });

    test('Test toggle button', () => {
        render(<App/>);
        const toggleButton = screen.getByTestId('toggle-button-id');
        expect(toggleButton).toBeInTheDocument();
        expect(screen.queryByTestId('toggle-div-id')).toBeNull();
        fireEvent.click(toggleButton);
        expect(screen.queryByTestId('toggle-div-id')).toBeInTheDocument();
        fireEvent.click(toggleButton);
        expect(screen.queryByTestId('toggle-div-id')).toBeNull();
    });

    test('Test input fire event', () => {
        render(<App/>);
        const input = screen.getByTestId('input-id');
        expect(input).toBeInTheDocument();

        const inputValue = screen.getByTestId('input-value');
        expect(inputValue).toBeInTheDocument();
        expect(inputValue).toContainHTML('');
        fireEvent.input(input, {
            target: {value: 'vadmark'}
        });
        // screen.debug();
        expect(inputValue).toContainHTML('vadmark');
    });

    test('Test input user event', async () => {
        render(<App/>);
        const input = screen.getByTestId('input-id');
        expect(input).toBeInTheDocument();

        const inputValue = screen.getByTestId('input-value');
        expect(inputValue).toBeInTheDocument();
        expect(inputValue).toContainHTML('');
        await userEvent.type(input, 'vadmark');
        // screen.debug();
        expect(inputValue).toContainHTML('vadmark');
    });
});

