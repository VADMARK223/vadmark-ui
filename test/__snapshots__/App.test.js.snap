// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Basic test 1`] = `
<input
  data-testid="input-id"
  placeholder="input value.."
  type="text"
/>
`;
